simulation_ws/src/test_nodes/nodes/navigation_test

```diff
@@ -0,0 +1,119 @@
+            self.is_cancelled = True
+            rospy.loginfo("Successfully requested cancel job")
+            self.set_tag(name=self.test_name + "_Time_Elapsed_End", value= str(time.time()).split(".", 1)[0])
+        else:
+            rospy.logerr("Cancel request failed: %s", response.message)
+    
+    def set_tag(self, name, value):
+        rospy.wait_for_service("/robomaker/job/add_tags")
+        requestAddTags = rospy.ServiceProxy("/robomaker/job/add_tags", AddTags)
+        tags = ([Tag(key=name, value=value)])
+        response = requestAddTags(tags)
+        if response.success:
+            rospy.loginfo("Successfully added tags: %s", tags)
+        else:
+            rospy.logerr("Add tags request failed for tags (%s): %s", tags, response.message)
+
+    def setUp(self):
+        self.latch = False
+        self.successful_navigations = 0
+        self.test_name = "Robot_Monitoring_Tests_" + str(time.time()).split(".", 1)[0]
+        self.is_completed = False
+        rospy.loginfo("Test Name: %s", self.test_name)
+        self.navigation_success_count = rospy.get_param('NAVIGATION_SUCCESS_COUNT')
+        self.timout = rospy.get_param("CLOCK_TIMEOUT_SECONDS")
+        
+    def set_latched(self):
+        self.latch = True
+    
+    def set_unlatched(self):
+        self.latch = False
+    
+    def increment_navigations(self):
+        self.successful_navigations = self.successful_navigations + 1
+    
+    def is_complete(self):
+        return self.successful_navigations >= self.navigation_success_count
+        
+    def check_timeout(self, msg):
+        """
+            Cancel the test if it times out. The timeout is based on the
+            /clock topic (simulation time).
+        """
+        if msg.clock.secs > self.timout and not self.is_cancelled:
+            rospy.loginfo("Test timed out, cancelling job")
+            self.set_tag(name=self.test_name + "_Status", value="Failed")
+            self.set_tag(name=self.test_name + "_Timed_Out", value=str(self.timout))
+            self.cancel_job()
+            
+    def check_complete(self, msgs):
+        for msg in msgs.metrics:
+            if msg.metric_name == METRIC_NAME:
+                rospy.loginfo("Metric Name: %s", msg.metric_name)
+                rospy.loginfo("Metric Value: %s", msg.value)
+                """
+                    If our distance to goal metric drops below 1 and we've
+                    achieved our goal count then we are complete and we tag the
+                    job success and cancel. Else, we continue checking progress
+                    towards a new goal once the distance to goal climbs back 
+                    above 1. Note that we're using what the nav stack thinks
+                    is the distance to goal, in the real world we'd want to use
+                    a ground truth value to ensure accuracy. 
+                """
+                if msg.value <= 1 and self.is_completed == False and self.latch == False:
```

Comment:
be aware the value could fluctuate around 1. Should use a range (which you had before but assumed it triggered at == 0.0)